zappa/core.py

```diff
@@ -2098,12 +2098,17 @@ def get_rest_apis(self, project_name):
         """
         Generator that allows to iterate per every available apis.
         """
-        all_apis = self.apigateway_client.get_rest_apis(limit=500)
+        pagination_position = "init"
 
-        for api in all_apis["items"]:
-            if api["name"] != project_name:
-                continue
-            yield api
+        while pagination_position:
+            if pagination_position != "init":
+                response = self.apigateway_client.get_rest_apis(limit=500, position=pagination_position)
+            else:
+                response = self.apigateway_client.get_rest_apis(limit=500)
+
+            yield from [api for api in response.get("items", []) if api["name"] == project_name]
```

Comment:
This can be be a generator expression in place of a list-comprehension. (generator expression will evaluate and return one by one as opposed to instantiating a list then yielding objects from that list.
```
yield from (api for api in response.get("items", []) if api["name"] == project_name)
```